// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe.autoflip;

import "mediapipe/examples/desktop/autoflip/autoflip_messages.proto";
import "mediapipe/framework/calculator.proto";

// Next tag: 3
message ShotChangeFusingCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional ShotChangeFusingCalculatorOptions ext = 280092317;
  }
  // Setting related to each type of signal this calculator could process.
  repeated ShotSettings shot_settings = 1;

  // If two shot change signals are within min_shot_span(in seconds),
  // they are considred as one siganl.
  optional double min_shot_span = 2 [default = 0.0];
}

// Next tag: 3
message ShotSettings {
  // For tag interface, optional for order interface.
  optional int32 id = 1 [default = -1];

  // Priority of the shot change signal. static shot (scene) change has the 
  // hightest priority 0, since we need to precess the scene whenever there
  // is a static shot change.
  optional int32 priority = 2 [default = 0];
}
